color/quoted.c

```diff
@@ -122,3 +122,402 @@ bool quoted_colors_parse_color(enum ColorId color, uint32_t fg, uint32_t bg,
   *rc = MUTT_CMD_SUCCESS;
   return true;
 }
+
+/**
+ * qclass_free - Free a single QClass
+ * @param ptr QClass to free
+ */
+static void qclass_free(struct QClass **ptr)
+{
+  if (!ptr || !*ptr)
+    return;
+
+  struct QClass *qc = *ptr;
+
+  FREE(&qc->prefix);
+  FREE(ptr);
+}
+
+/**
+ * qclass_free_tree - Free a tree of QClass
+ * @param[out] quote_list Quote list to free
+ */
+void qclass_free_tree(struct QClass **quote_list)
```

Comment:
Are there benefits to naming this `qclass_free_tree()` instead of `qclass_free()`? It breaks the expected pattern of `<struct>_free()` for cleanup that the codebase follows. I can see contributors calling `qclass_free()` by mistake.

nit: the `QClass` uses a bidirectional graph instead of a tree.

Reply:
It means that `_new()` and `_free()` are equal and opposite.
I've added `@note`s emphasising the difference between `_free()` and `_free_tree()`